include/dlaf/matrix/matrix_base.h

```diff
@@ -33,10 +49,15 @@ class MatrixBase {
   }
 
   /// Returns the block size of the matrix.
-  const TileElementSize& blockSize() const noexcept {
+  const TileElementSize blockSize() const noexcept {
```

Comment:
Is the reference gone intentionally? If yes, why not:
```suggestion
  TileElementSize blockSize() const noexcept {
```
?